internal/utils/endorsement.go

```diff
@@ -0,0 +1,42 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/fabric-protos-go/peer/lifecycle"
+	"github.com/hyperledger/fabric/bccsp/factory"
+	"github.com/hyperledger/fabric/common/cauthdsl"
+	"github.com/hyperledger/fabric/msp"
+	"github.com/hyperledger/fabric/msp/mgmt"
+)
+
+func NewPolicyEvaluator() *PolicyEvaluator {
+	manager := mgmt.NewDeserializersManager(factory.GetDefault())
+	return &PolicyEvaluator{
+		IdentityDeserializer: manager.GetLocalDeserializer(),
+	}
+}
+
+type PolicyEvaluator struct {
+	msp.IdentityDeserializer
+}
+
+func (id *PolicyEvaluator) EvaluateIdentity(policyBytes []byte, identityBytes []byte) error {
+	identity, err := id.IdentityDeserializer.DeserializeIdentity(identityBytes)
+	if err != nil {
+		return fmt.Errorf("error while deserialzing identity, err: %s", err)
+	}
+
+	pp := cauthdsl.NewPolicyProvider(id.IdentityDeserializer)
+	policy, _, err := pp.NewPolicy(policyBytes)
+	if err != nil {
+		return err
+	}
+
+	return policy.EvaluateIdentities([]msp.Identity{identity})
+}
+
+func CanEndorse(identityBytes []byte, ccDef *lifecycle.QueryApprovedChaincodeDefinitionResult) error {
```

Comment:
This seems a slightly different version of canEndorse from the PUMLs.  In that case we do it on the PK, not the (peer) identity.  

I realize, though, also that the canEndorse in the PUML is not completely consistent in the PUMLS. There is actually a slighly different case for the use in ercc during registration and ecc during key dist.  In the former (and contrary to what the comment says) we will call it _before_ registration in ercc, so (but steps before canEndorse ensure it is of proper verified mrenclave etc) whereas in the ecc case it the registration check is crucial as it transitively ensures validation of mrenclave etc.   Given that you currently use it for ercc your current implementation i guess is the only thing you need. That said, maybe call it slightly differently to not confuse with the (semantically different) `can_endorse` from the puml?

Reply:
you are right, we better rename this function and let ercc implement `canEndorse` function as described in the PUMLs. Here we should only focus on the primitives that are needed to implement `canEndorse`. I guess this function here is more useful Org-Enclave binding.

Reply:
renamed.